src/nvim/charset.c

```diff
@@ -1743,16 +1743,16 @@ int vim_isblankline(char_u *lbuf)
 /// @param nptr Returns the signed result.
 /// @param unptr Returns the unsigned result.
 void vim_str2nr(char_u *start, int *hexp, int *len, int dooct, int dohex,
-                long *nptr, unsigned long *unptr)
+                intmax_t *nptr, uintmax_t *unptr)
```

Comment:
By the way, since signature is changed it makes sense to add `FUNC_ATTR_NONNULL_ARG(1)`.